### DISCUSSION TITLE
Sophisticated window switcher

### DISCUSSION THREAD

**Discussion by wolfisdev** [qtile/qtile#2428] [Q&A] [answered]
I'm looking for functionality similar to KDE's **present all windows** feature or the **parachute** KWIN script.

Those zoom out and show you (previews of) all open windows (of all desktops/tags) at once. Then you can start typing and it will fuzzy filter the window list and when hitting enter it will switch to the desktop/tag of the selected window and focus on it.

This is great because you don't have to remember where you opened certain windows and can still very quickly and reliably focus them from anywhere.

What I found is kind of similar is rofi's window mode, but I'd prefer something that actually provides a preview of the windows as well, which makes it much easier to distinguish between windows of the same application (multiple firefox windows e.g.)

Is there something like this already around in the qtile realm?

Thanks for your time.

**Comment by m-col** [comment, marked as answer]
The short answer is no. These kinds of features are generally handled by a compositing manager; KWIN is a compositing window manager. Qtile just does window management and not compositing so can't do it. However, there is nothing stopping Qtile from being run alongside a standalone compositing manager that has this feature, though I don't know of any. For some reason there aren't many standalone compositing managers.

However, I have heard stories of people setting up standalone window managers such as Qtile to run within full desktop environments like gnome, KDE etc, so maybe it's possible to run Qtile inside KDE (or just within KWIN... not sure to what extent KWIN requires KDE) and disable KWIN's window management and just use its compositing. It might be tricky but might be worth looking into.

**Reply by wolfisdev** [reply]
Thanks for the pointers.

Is a compositing manager the same as a compositor,  like *compton* or *picom*?

I'm running KDE as my main system and there is a very good dynamic tiling kwin script called *krohnkite* that basically provides the best of both worlds (depending on your point of view). But KDE has been pretty unreliable and keeps breaking on me lately and since I'm not very good at fixing such stuff at some point I can't be bothered anymore and am strongly considering switching to qtile or something similar.

**Reply by m-col** [reply]
> Is a compositing manager the same as a compositor, like compton or picom?

Yep exactly. I guess what you're after would be Qtile but with kwin's compositing running on top so you can get the nice overviews and other fancy scripts working. I'm sure someone somewhere will have got it working, not necessarily with Qtile, but the same approach would be possible with other tiling WMs like i3, so also checking through forums for other WMs you might find something useful.

**Reply by wolfisdev** [reply]
Cool, I'll give it a go. Thanks.
